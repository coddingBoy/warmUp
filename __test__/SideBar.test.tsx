import React from 'react'
import axios from 'axios'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import SideBar from '../src/containers/SideBar'

Enzyme.configure({ adapter: new Adapter() })
jest.mock('axios')

test('render', () => {
    const component = renderer.create(<SideBar getIsReset={jest.fn()} getRoutes={jest.fn()} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

test('submit success', () => {
    const mockGetIsReset = jest.fn()
    const mockGetRoutes = jest.fn()
    axios.post.mockResolvedValue({
        data: { token: '8b01b5646c16' },
    })
    axios.get.mockResolvedValue({
        data: {
            status: 'success',
            path: [
                ['22.372081', '114.107877'],
                ['22.326442', '114.167811'],
                ['22.284419', '114.159510'],
            ],
            total_distance: 20000,
            total_time: 1800,
        },
    })
    const component = shallow(<SideBar getIsReset={mockGetIsReset} getRoutes={mockGetRoutes} />)
    component.find('#submit').simulate('click')
    expect(mockGetIsReset.mock.calls.length).toBe(1)
    setTimeout(() => {
        expect(mockGetRoutes.mock.calls.length).toBe(1)
        expect(component.state().totalDistance).toBe(20000)
        expect(component.state().totalTime).toBe(1800)
        expect(component.state().error).toBe('')
    })
})

test('submit fail', () => {
    const mockGetIsReset = jest.fn()
    const mockGetRoutes = jest.fn()
    axios.post.mockResolvedValue({
        data: { token: '8b01b5646c16' },
    })
    axios.get.mockResolvedValue({
        data: {
            status: 'failure',
            error: 'Location not accessible by car',
        },
    })
    const component = shallow(<SideBar getIsReset={mockGetIsReset} getRoutes={mockGetRoutes} />)
    component.find('#submit').simulate('click')
    expect(mockGetIsReset.mock.calls.length).toBe(1)
    setTimeout(() => {
        expect(mockGetRoutes.mock.calls.length).toBe(0)
        expect(component.state().totalDistance).toBe(0)
        expect(component.state().totalTime).toBe(0)
        expect(component.state().error).toBe('Location not accessible by car')
    })
})

test('reset', () => {
    const mockGetIsReset = jest.fn()
    const mockGetRoutes = jest.fn()
    const component = shallow(<SideBar getIsReset={mockGetIsReset} getRoutes={mockGetRoutes} />)
    component.find('#reset').simulate('click')
    expect(mockGetIsReset.mock.calls.length).toBe(1)
    setTimeout(() => {
        expect(component.state().totalDistance).toBe(0)
        expect(component.state().totalTime).toBe(0)
        expect(component.state().error).toBe('')
    })
})
