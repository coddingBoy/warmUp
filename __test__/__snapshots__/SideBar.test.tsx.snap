// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="sidebar"
>
  <ul
    className="container"
  >
    <li>
      <input
        autoComplete="on"
        onChange={[Function]}
        placeholder="Starting location"
        value=""
      />
      
    </li>
    <li>
      <input
        onChange={[Function]}
        placeholder="Drop-off point"
        value=""
      />
      
    </li>
  </ul>
  <div
    className="buttonGroup"
  >
    <button
      id="submit"
      onClick={[Function]}
    >
      submit
    </button>
    <button
      id="reset"
      onClick={[Function]}
    >
      reset
    </button>
  </div>
</div>
`;
